dbms/src/Parsers/ASTSampleRatio.h

```diff
@@ -12,7 +11,11 @@ namespace DB
 class ASTSampleRatio : public IAST
 {
 public:
+#ifdef __SIZEOF_INT128__
     using BigNum = __uint128_t;    /// Must contain the result of multiplying two UInt64.
+#else
+    using BigNum = uint64_t;
```

Comment:
Missing comment that this fallback is incomplete.

Reply:
Better to add a header, which defines `uint128_t` to `__uint128_t` if available and use `boost::multiprecision` as a fallback.